src/ai/backend/agent/agent.py

```diff
@@ -927,6 +928,39 @@ async def _handle_clean_event(self, ev: ContainerLifecycleEvent) -> None:
                     if ev.done_future is not None and not ev.done_future.done():
                         ev.done_future.set_result(None)
 
+    @abstractmethod
+    async def pause_kernel(
+        self,
+        kernel_id: KernelId,
+        container_id: Optional[ContainerId],
+    ) -> None:
+        pass
+
+    @abstractmethod
+    async def unpause_kernel(
+        self,
+        kernel_id: KernelId,
+        container_id: Optional[ContainerId],
+    ) -> None:
+        pass
```

Comment:
Please implement these methods on Kubernetes backend also. If you don't have any knowledge about Kubernetes, it's okay to let implementations as blank and create separate issue stating we have to implement it later.